activityMonitor.js

```diff
@@ -0,0 +1,48 @@
+const { exec } = require("child_process");
+const fs = require("fs");
+
+let lastLog = Date.now();
+const ONE_MINUTE = 60000;
+const INTERVAL_TEN_SECONDS = 100;
+const LOG_FILE_NAME = "activityMonitor.log";
+const SCRIPT_FOR_WINDOWS = `powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + ' ' + $_.CPU + ' ' + $_.WorkingSet }"`;
+const SCRIPT_ALL_PLATFORM = "ps -A -o %cpu,%mem,comm | sort -nr | head -n 1";
+
+setInterval(() => {
+  const command =
+    process.platform === "win32" ? SCRIPT_FOR_WINDOWS : SCRIPT_ALL_PLATFORM;
+
+  exec(command, (error, stdout, stderr) => {
+    if (error) {
+      console.log(`Error: ${error.message}`);
+      return;
+    }
+    if (stderr) {
+      console.log(`Error: ${stderr}`);
+      return;
+    }
+
+    const log = getLog(stdout);
+
+    process.stdout.write("\r" + " ".repeat(process.stdout.columns - 1) + "\r");
+
+    process.stdout.write(log);
+
+    const now = Date.now();
+    if (now - lastLog >= ONE_MINUTE) {
+      fs.appendFile(LOG_FILE_NAME, `<${new Date(now)}> : ${log}\n`, (err) => {
```

Comment:
better to use appendFileSync and such operations better to wrap in try catch block

Reply:
Why is it better to use a synchronous operation here instead of an asynchronous one?

Reply:
With this example it's doesn't matter but with huge amount of better to ensure this operation is done before next execution of code. In this example it's not noticeable at all